main.go

```diff
@@ -147,8 +153,8 @@ func main() {
 		defer os.Exit(1)
 	}
 	// create a timeout
-	ctx, cancel = context.WithTimeout(ctx, 5*time.Second)
-	defer cancel()
+	ctx, cancelHttpServerCtx := context.WithTimeout(ctx, 5*time.Second)
```

Comment:
`cancelHTTPCtx` is good enough I think. Also HTTP, because https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Reply:
Sounds good. I haven't read that wiki in a few months - it was a good refresher.